tests/back/api/test_explainers_api.py

```diff
@@ -0,0 +1,147 @@
+        dataset_id=dataset_id,
+        task_name="TabularClassificationTask",
+        name="Test Experiment",
+    )
+    db.add(dummy_experiment)
+    db.commit()
+    db.refresh(dummy_experiment)
+    yield dummy_experiment.id
+
+    db.delete(dummy_experiment)
+    db.commit()
+
+
+@pytest.fixture(scope="module", name="run_id")
+def fixture_run_id(session: sessionmaker, experiment_id: int):
+    db = session()
+    run = Run(
+        experiment_id=experiment_id,
+        model_name="RandomForestClassifier",
+        parameters={},
+        name="Run",
+    )
+    db.add(run)
+    db.commit()
+    db.refresh(run)
+
+    yield run.id
+
+    db.delete(run)
+    db.commit()
+    db.close()
+
+
+def test_create_explainer(client: TestClient, run_id: int):
+    response = client.post(
+        "/api/v1/explainer/",
+        json={
+            "name": "test_explainer",
+            "run_id": run_id,
+            "explainer_name": "PartialDependence",
+            "parameters": {"grid_resolution": 100, "percentiles": [0.1, 0.6]},
+        },
+    )
+    assert response.status_code == 201, response.text
+
+
+def test_get_explainer(client: TestClient, run_id: int, dataset_id: int):
+    response = client.get("/api/v1/explainer/1")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data["name"] == "test_explainer"
+    assert data["run_id"] == run_id
+    assert data["dataset_id"] == dataset_id
+    assert data["explainer_name"] == "PartialDependence"
+    assert data["explainer_path"] == os.path.join(
+        settings.USER_EXPLAINER_PATH, f"{1}.pkl"
+    )
+
+
+def test_get_explainers(client: TestClient, run_id: int, dataset_id: int):
+    response = client.get("/api/v1/explainer")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data[0]["name"] == "test_explainer"
+    assert data[0]["run_id"] == run_id
+    assert data[0]["dataset_id"] == dataset_id
+    assert data[0]["explainer_name"] == "PartialDependence"
+    assert data[0]["explainer_path"] == os.path.join(
+        settings.USER_EXPLAINER_PATH, f"{1}.pkl"
+    )
+    response = client.get("/api/v1/explainer/?run_id=1")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data[0]["name"] == "test_explainer"
+    assert data[0]["run_id"] == run_id
+    assert data[0]["dataset_id"] == dataset_id
+    assert data[0]["explainer_name"] == "PartialDependence"
+    assert data[0]["explainer_path"] == os.path.join(
+        settings.USER_EXPLAINER_PATH, f"{1}.pkl"
+    )
+
+
+def test_get_wrong_explainer(client: TestClient):
+    response = client.get("/api/v1/explainer/666")
+    assert response.status_code == 404, response.text
+    assert response.text == '{"detail":"Explainer not found"}'
+
+
+def test_delete_explainer(client: TestClient):
+    response = client.delete("/api/v1/explainer/1")
+    assert response.status_code == 204, response.text
```

Comment:
Is there any case where an explainer cannot be removed?

For example, in a case where the explainer has associated explanations? In that case, it would be necessary to decide whether to cascade deletion or not to allow deletion until the dependent objects are deleted.